python/ray/util/sgd/v2/worker_group.py

```diff
@@ -220,3 +243,17 @@ def add_workers(self, num_workers: int):
 
     def __len__(self):
         return len(self.workers)
+
+
+def create_executable_worker_group(executable_cls: Type, *args,
+                                   **kwargs) -> WorkerGroup:
+    """Create a WorkerGroup from the provided ``executable_cls``.
+
+    This function will automatically handle subclassing BaseWorker.
+    """
+
+    class _WrappedExecutable(executable_cls, BaseWorkerMixin):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+    return WorkerGroup(*args, actor_cls=_WrappedExecutable, **kwargs)
```

Comment:
I'm wondering if we can do one of the following:
1. Move this logic of creating the new class  directly into `WorkerGroup.__init__`.
2. Change this to only create and return the class and not the `WorkerGroup`. 
3. Pass in the `actor_cls_args` and `actor_cls_kwargs` here.

Reason being that right now it seems like we're exposing a new path to create an executable `WorkerGroup` but most of the logic is still in the generic `__init__()`.

Reply:
Good point, changed this so it's handled inside WorkerGroup.__init__